function/pkg/operator/generic_operator.go

```diff
@@ -38,13 +39,17 @@ func (p genericOperator) Apply(ctx context.Context, opts ApplyOptions) error {
 	return nil
 }
 
-func (p genericOperator) apply(ctx context.Context, item unstructured.Unstructured, opts ApplyOptions) (*unstructured.Unstructured, client.PostStatusEntry, error) {
+func (p genericOperator) apply(ctx context.Context, item unstructured.Unstructured,
+	opts ApplyOptions) (*unstructured.Unstructured, client.PostStatusEntry, error) {
 	applied, statusEntry, err := applyObject(ctx, p.Client, item, opts.DryRun)
 	if err != nil {
 		return applied, statusEntry, err
 	}
 	if opts.WaitForApply {
 		err = waitForObject(ctx, p.Client, *applied)
+		if err != nil {
+			return applied, statusEntry, err
+		}
```

Comment:
Just wondering if this is necessary? Shouldn't it set the existing owner err variable which is returned below anyway?

Reply:
yes but Checkmarx detected that this is incorrect error handling, that's why I had to add it here